stix_shifter/stix_translation/src/patterns/grammar/STIXPatternParser.py

```diff
@@ -1596,6 +1596,12 @@ def TimestampLiteral(self, i:int=None):
         def STOP(self):
             return self.getToken(STIXPatternParser.STOP, 0)
 
+        def start_time(self):
```

Comment:
I don't know if it's a good idea to add code directly to this file since it is generated from the STIXPattern.g4 file. The grammar files get generated by running 
`antlr4 -Dlanguage=Python3 -listener -visitor STIXPattern.g4`
Even though the g4 file is included in this project it's mostly a copy of https://github.com/oasis-open/cti-stix2-json-schemas/blob/master/pattern_grammar/STIXPattern.g4. I say mostly because it appears the g4 file used in the original Mitre translation project was not from that oasis repo. There are differences and a straight copy breaks a bunch of tests.

Reply:
Ah good catch, I didn't see that it was generated. I'll investigate this a bit further. I do think this should be done in a way so each parser doesn't need to duplicate a regex for the timestamp logic though.